validphys2/src/validphys/photon/compute.py

```diff
@@ -0,0 +1,319 @@
+"""Script that calls fiatlux to add the photon PDF."""
+import numpy as np
+from scipy.interpolate import interp1d
+from scipy.integrate import trapezoid
+from os import remove
+import time
+import sys
+
+import fiatlux
+import yaml
+from eko.io import EKO
+
+# from eko.io.manipulate import xgrid_reshape
+# from eko.interpolation import XGrid
+
+from validphys.lhapdfset import LHAPDFSet
+from validphys.n3fit_data import replica_nnseed
+from . import structure_functions as sf
+import logging
+from n3fit.io.writer import XGRID
+
+
+log = logging.getLogger(__name__)
+
+
+class Photon:
+    """Photon class computing the photon array with the LuxQED approach."""
+
+    def __init__(self, theoryid, fiatlux_runcard, replicas_id):
+        self.theory = theoryid.get_description()
+        self.fiatlux_runcard = fiatlux_runcard
+        self.replicas_id = replicas_id
+        self.q_in = 100
+        self.q_in2 = self.q_in**2
+
+        # parameters for the alphaem running
+        self.alpha_em_ref = self.theory["alphaqed"]
+        self.qref = self.theory["Qref"]
+
+        self.set_betas()
+        self.set_thresholds_alpha_em()
+
+        # structure functions
+        self.qcd_pdfs = LHAPDFSet(fiatlux_runcard["pdf_name"], "replicas")
+
+        # TODO : maybe find a different name for fiatlux_dis_F2
+        path_to_F2 = theoryid.path / "fastkernel/fiatlux_dis_F2.pineappl.lz4"
+        path_to_FL = theoryid.path / "fastkernel/fiatlux_dis_FL.pineappl.lz4"
+        self.path_to_eko_photon = theoryid.path / "eko_photon.tar"
+
+        # set fiatlux
+        self.lux = {}
+        f2 = {}
+        fl = {}
+        f2lo = {}
+        for id in replicas_id:
+            f2[id] = sf.InterpStructureFunction(path_to_F2, self.qcd_pdfs.members[id])
+            fl[id] = sf.InterpStructureFunction(path_to_FL, self.qcd_pdfs.members[id])
+            f2lo[id] = sf.F2LO(self.qcd_pdfs.members[id], self.theory)
+            ff = open(f"fiatlux_runcard_{id}.yml", "w+")
+            yaml.dump(self.fiatlux_runcard, ff)
+            self.lux[id] = fiatlux.FiatLux(f"fiatlux_runcard_{id}.yml")
+            remove(f"fiatlux_runcard_{id}.yml")
```

Comment:
```suggestion
            with open(f"fiatlux_runcard_{id}.yml", "w+") as ff:
                yaml.dump(self.fiatlux_runcard, ff)
                self.lux[id] = fiatlux.FiatLux(f"fiatlux_runcard_{id}.yml")
                remove(f"fiatlux_runcard_{id}.yml")
```

Reply:
```suggestion
            with open(f"fiatlux_runcard_{id}.yml", "w+") as ff:
                yaml.dump(self.fiatlux_runcard, ff)
            self.lux[id] = fiatlux.FiatLux(f"fiatlux_runcard_{id}.yml")
            remove(f"fiatlux_runcard_{id}.yml")
```
I guess you meant like this, no? else looks like you're removing the file you're writing

Reply:
Ah yes indeed